### PR TITLE
Fix Windows long path issues

### PR DESCRIPTION
**Pull request by Kawzeg**
Taken from https://github.com/haskell/cabal/pull/4977/files

Fixes typelead/eta#691

### CODE REVIEW

**Comment by rahulmutt** [comment]
@jneira Can you take a look at this?

**Comment by jneira** [comment]
@rahulmutt sure!
@Kawzeg thank very much, this will do the windows experience much better

File: etlas-cabal/etlas-cabal.cabal [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name:          etlas-cabal
 -- @VERSION
-version:       1.4.0.1
```
Comment (jneira) [inline etlas-cabal/etlas-cabal.cabal:3]: Not sure if a new version is needed cause it usually involves update the binaries in downloading servers, right @rahulmutt?

Comment (rahulmutt) [reply etlas-cabal/etlas-cabal.cabal:3]: The binaries are manually uploaded - we haven't setup an automated CI system yet, so this is OK.

**Review by rahulmutt — APPROVED** [review]
Thanks for the usability improvement on Windows, @Kawzeg!

**Comment by jneira** [comment]
Sorry for not update the pr but yesterday at late night i tested the pr in win 10 and i got errors even for short paths. I want to test it in another win version to be sure though.
It works for cabal 2.2 so it *should* work for etlas but maybe other changes of cabal makes it work.
@Kawzeg did it work for you in some windows env?

**Comment by rahulmutt** [comment]
Reverted in #53. @Kawzeg Can you take a look and see if you can fix the issue and submit a new PR?

**Comment by Kawzeg** [comment]
@jneira 
I built this on my Win 7 machine because the long path issue prevented me from building some dependency, and it worked for me, though I didn't test it extensively and may have mixed up some changes. I'll try if I can get it to work on win 10.

@rahulmutt I will try!

**Comment by jneira** [comment]
@Kawzeg It works for me in win 7 too:
* Setting the `store-dir` with a path around 50 (it fails for 60). Commenting out the property works too cause the default store-dir in `%APPDATA%\etlas\store` is 37 for me.
* The previous version fails with paths around 30
* So hopefully the new version will work for most windows users *without the workaround of setting the `store-dir` prop* to a shorter path
* I guess the project being built will need to have a path in those limits, taking in account the dirs created in `dist`

In win 10 i've tested it with `store-dir: D:\sd` in `%APPDATA%\etlas\config` file and it throwed errors related with handling the `package-db` dir, so maybe change that property is related with the error. I will try again later from scratch.

**Comment by rahulmutt** [comment]
@Kawzeg Your PR was good. @jneira verified that stale build artifacts and already existing bugs were the cause and not your changes. I've undone the revert in #54.

**Comment by jneira** [comment]
Yeah, @Kawzeg, sorry for the false positive and thanks again for the fix

**Pull request merged**